sympy/vector/functions.py

```diff
@@ -196,24 +216,68 @@ def gradient(scalar, coord_sys):
     scalar : SymPy Expr
         The scalar field to compute the gradient of
 
-    coord_sys : CoordSysCartesian
-        The coordinate system to calculate the gradient in
-
     Examples
     ========
 
-    >>> from sympy.vector import CoordSysCartesian, gradient
-    >>> R = CoordSysCartesian('R')
+    >>> from sympy.vector import CoordSystem3D, gradient
+    >>> R = CoordSystem3D('R')
     >>> s1 = R.x*R.y*R.z
-    >>> gradient(s1, R)
+    >>> gradient(s1)
     R.y*R.z*R.i + R.x*R.z*R.j + R.x*R.y*R.k
     >>> s2 = 5*R.x**2*R.z
-    >>> gradient(s2, R)
+    >>> gradient(s2)
     10*R.x*R.z*R.i + 5*R.x**2*R.k
 
     """
+    # Check input
+    if isinstance(scalar, Zero) or scalar == 0:
+        return Vector.zero
+
+    if not issubclass(type(scalar), Expr) or scalar.is_Vector:
+        raise TypeError("Your input should be a scalar field.")
+
+    for atom in scalar.atoms():
+        if isinstance(atom, BaseScalar):
+            system = atom.system
+            break
```

Comment:
I agree. If we really want to have every vector be defined in only _one_ coordinate system, then its better to remove the coordinate system parameter from this function as this PR does it. Otherwise, it would be better to leave it in.